src/adapters/superFluid/superFluidAdapter.ts

```diff
@@ -167,4 +183,54 @@ export class SuperFluidAdapter implements SuperFluidAdapterInterface {
       logger.error(e);
     }
   }
+
+  async getFlowByTxHash(params: {
+    receiver: string;
+    sender: string;
+    flowRate: string;
+    transactionHash: string;
+  }): Promise<FlowUpdatedEvent | undefined> {
+    try {
+      const response = await axios.post(subgraphUrl, {
+        query: getFlowsQuery,
+        variables: {
+          where: params,
+          orderBy: 'timestamp',
+          orderDirection: 'asc',
+        },
+      });
+      const flowUpdates = response.data?.data
+        ?.flowUpdatedEvents as FlowUpdatedEvent[];
+      return flowUpdates?.[0];
+    } catch (e) {
+      logger.error('getFlowByReceiverSenderFlowRate error', e);
+      throw e;
+    }
+  }
+
+  async getFlowByReceiverSenderFlowRate(params: {
+    receiver: string;
+    sender: string;
+    flowRate: string;
+    timestamp_gt: number;
+  }): Promise<FlowUpdatedEvent | undefined> {
+    try {
+      logger.debug('getFlowByReceiverSenderFlowRate has been called', params);
+
+      const response = await axios.post(subgraphUrl, {
+        query: getFlowsQuery,
+        variables: {
+          where: params,
+          orderBy: 'timestamp',
+          orderDirection: 'asc',
+        },
+      });
+      const flowUpdates = response.data?.data
+        ?.flowUpdatedEvents as FlowUpdatedEvent[];
+      return flowUpdates?.[0];
```

Comment:
will this crash if flowUpdates is empty

Reply:
I tested it will not crash neither `flowUpdates` is empty nor undefined

<img width="466" alt="Screenshot 1403-01-28 at 5 58 07 in the afternoon" src="https://github.com/Giveth/impact-graph/assets/9850545/9df21ece-00be-4520-ba8c-715f22463748">